pkg/store/bucket.go

```diff
@@ -1254,7 +1254,7 @@ func debugFoundBlockSetOverview(logger log.Logger, mint, maxt, maxResolutionMill
 
 // Series implements the storepb.StoreServer interface.
 func (s *BucketStore) Series(req *storepb.SeriesRequest, seriesSrv storepb.Store_SeriesServer) (err error) {
-	srv := newFlushableServer(seriesSrv, s.LabelNamesSet(), req.WithoutReplicaLabels)
+	srv := newFlushableServer(seriesSrv)
```

Comment:
Since we already use a proxy-heap in BucketStore, can we re-sort at the block level instead of globally?

Reply:
Yeah! that was possible without breaking any tests! But we need to use eagerSeriesSet for that to work.